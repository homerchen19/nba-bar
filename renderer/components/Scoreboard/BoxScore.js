import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';
import { StickyTable, Row, Cell } from 'react-sticky-table';

import { colors } from '@styles/theme';
import nba from '@utils/nba';

const Wrapper = styled(Flex)`
  width: 100%;
  height: 300px;
  overflow-y: scroll !important;

  > .am-tabs * {
    box-sizing: content-box;
  }
`;

const HeaderCell = styled(Cell)`
  padding: 6px;
  background: ${colors.blue};
  color: #fff;
  text-align: center;
`;

const StyledCell = styled(Cell)`
  padding: 6px;
  border-bottom: 1px solid ${colors.white};
  background: #fff;
  text-align: center;
`;

const Button = styled.button`
  width: 30%;
  height: 28px;
  margin: 8px 0;
  border: 1px solid ${colors.white};
  opacity: 1;
  background-color: ${props =>
    props.selected ? props.background : colors.white} !important;
  color: ${props => (props.selected ? '#fff' : colors.black)} !important;
  transition: all ease 0.2s;
  cursor: pointer;

  & :hover {
    ${props =>
      props.selected
        ? 'opacity: 0.9;'
        : `
    color: ${colors.white} !important;
    background-color: ${props.background} !important;
    `};
  }
`;

const cells = [
  'Player',
  'MIN',
  'PTS',
  'REB',
  'AST',
  'STL',
  'BLK',
  'FG',
  '3P',
  'FT',
  'TO',
  '+/-',
];

const renderHeaderCell = () =>
  cells.map(cell => (
    <HeaderCell key={cell}>
      <b>{cell}</b>
    </HeaderCell>
  ));

const renderTeamBoxScoreTable = team => (
  <StickyTable>
    <Row key="header">{renderHeaderCell()}</Row>

    {team.players.map(player => (
      <Row key={`${player.first_name}${player.last_name}`}>
        <StyledCell
          key="Player"
          style={{
            borderRight: `1px solid ${colors.white}`,
            textAlign: 'left',
            fontWeight: '500',
          }}
        >
          {player.last_name === '' ? player.first_name : player.last_name}
        </StyledCell>
        <StyledCell key="MIN">{player.minutes}</StyledCell>
        <StyledCell key="PTS">{player.points}</StyledCell>
        <StyledCell key="REB">
          {+player.rebounds_defensive + +player.rebounds_offensive}
        </StyledCell>
        <StyledCell key="AST">{player.assists}</StyledCell>
        <StyledCell key="STL">{player.steals}</StyledCell>
        <StyledCell key="BLK">{player.blocks}</StyledCell>
        <StyledCell key="FG">
          {`${player.field_goals_made}-${player.field_goals_attempted}`}
        </StyledCell>
        <StyledCell key="3P">
          {`${player.three_pointers_made}-${player.three_pointers_attempted}`}
        </StyledCell>
        <StyledCell key="FT">
          {`${player.free_throws_made}-${player.free_throws_attempted}`}
        </StyledCell>
        <StyledCell key="TOV">{player.turnovers}</StyledCell>
        <StyledCell key="+/-">{player.plus_minus}</StyledCell>
      </Row>
    ))}
  </StickyTable>
);

class Boxscore extends Component {
  state = {
    team: this.props.home,
    selected: 'home',
  };

  toggleTeam = (team, selected) => {
    this.setState({
      team,
      selected,
    });
  };

  render() {
    const { team, selected } = this.state;
    const { home, visitor } = this.props;

    return (
      <Fragment>
        <Flex justify="center" style={{ background: '#fff' }}>
          <Button
            selected={selected === 'home'}
            background={nba.getTeamBackgroundColor(home.name)}
            onClick={() => {
              this.toggleTeam(home, 'home');
            }}
            style={{
              borderRadius: '5px 0 0 5px',
            }}
          >
            {home.name}
          </Button>
          <Button
            selected={selected === 'visitor'}
            background={nba.getTeamBackgroundColor(visitor.name)}
            onClick={() => {
              this.toggleTeam(visitor, 'visitor');
            }}
            style={{
              borderRadius: '0 5px 5px 0',
            }}
          >
            {visitor.name}
          </Button>
        </Flex>
        <Wrapper>{renderTeamBoxScoreTable(team)}</Wrapper>
      </Fragment>
    );
  }
}

Boxscore.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default Boxscore;
