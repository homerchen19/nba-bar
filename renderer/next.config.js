const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
