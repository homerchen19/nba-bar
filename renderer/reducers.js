import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
