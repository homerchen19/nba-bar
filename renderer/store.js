import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
