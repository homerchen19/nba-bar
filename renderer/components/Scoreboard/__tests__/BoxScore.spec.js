import React from 'react';
import renderer from 'react-test-renderer';

import BoxScore from '../BoxScore';

const realMath = Object.create(global.Math);

beforeAll(() => {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
});

afterAll(() => {
  global.Math = realMath;
});

it('be defined', () => {
  expect(BoxScore).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    home: {
      name: 'IND',
      players: [
        {
          assists: '2',
          blocks: '1',
          field_goals_attempted: '13',
          field_goals_made: '11',
          first_name: 'Bojan',
          fouls: '3',
          free_throws_attempted: '0',
          free_throws_made: '0',
          jersey_number: '44',
          last_name: 'Bogdanovic',
          minutes: '30',
          on_court: '0',
          person_id: '202711',
          player_code: 'bojan_bogdanovic',
          plus_minus: '30',
          points: '28',
          position_full: 'Forward',
          position_short: 'F',
          rebounds_defensive: '3',
          rebounds_offensive: '0',
          seconds: '29',
          starting_position: 'SF',
          steals: '1',
          team_turnovers: '',
          three_pointers_attempted: '7',
          three_pointers_made: '6',
          turnovers: '2',
        },
      ],
    },
    visitor: {
      name: 'GSW',
      players: [
        {
          assists: '7',
          blocks: '1',
          field_goals_attempted: '23',
          field_goals_made: '8',
          first_name: 'Kevin',
          fouls: '3',
          free_throws_attempted: '9',
          free_throws_made: '9',
          jersey_number: '35',
          last_name: 'Durant',
          minutes: '32',
          on_court: '0',
          person_id: '201142',
          player_code: 'kevin_durant',
          plus_minus: '-17',
          points: '27',
          position_full: 'Forward',
          position_short: 'F',
          rebounds_defensive: '5',
          rebounds_offensive: '0',
          seconds: '56',
          starting_position: 'SF',
          steals: '3',
          team_turnovers: '',
          three_pointers_attempted: '10',
          three_pointers_made: '2',
          turnovers: '4',
        },
      ],
    },
  };
  const tree = renderer.create(<BoxScore {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
