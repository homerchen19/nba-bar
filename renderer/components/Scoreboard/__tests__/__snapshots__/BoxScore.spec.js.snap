// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
Array [
  <div
    className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
    style={
      Object {
        "background": "#fff",
      }
    }
  >
    <button
      className="BoxScore__Button-sc-1ktlrmv-3 dsuLua"
      onClick={[Function]}
      selected={true}
      style={
        Object {
          "borderRadius": "5px 0 0 5px",
        }
      }
    >
      IND
    </button>
    <button
      className="BoxScore__Button-sc-1ktlrmv-3 gOexTt"
      onClick={[Function]}
      selected={false}
      style={
        Object {
          "borderRadius": "0 5px 5px 0",
        }
      }
    >
      GSW
    </button>
  </div>,
  <div
    className="am-flexbox BoxScore__Wrapper-sc-1ktlrmv-0 GHxx am-flexbox-align-center"
  >
    <div
      className="sc-bxivhb iFCzUu sticky-table"
    >
      <div
        className="sc-bdVaJa iRVyCP sticky-table-table"
      >
        <div
          className="sc-htpNat ertYiC sticky-table-row"
        >
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              Player
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              MIN
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              PTS
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              REB
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              AST
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              STL
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              BLK
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              FG
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              3P
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              FT
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              TO
            </b>
          </div>
          <div
            className="sc-bwzfXH BoxScore__HeaderCell-sc-1ktlrmv-1 hqduNk sticky-table-cell"
          >
            <b>
              +/-
            </b>
          </div>
        </div>
        <div
          className="sc-htpNat ertYiC sticky-table-row"
        >
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
            style={
              Object {
                "borderRight": "1px solid #f7f7f7",
                "fontWeight": "500",
                "textAlign": "left",
              }
            }
          >
            Bogdanovic
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            30
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            28
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            3
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            2
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            1
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            1
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            11-13
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            6-7
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            0-0
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            2
          </div>
          <div
            className="sc-bwzfXH BoxScore__StyledCell-sc-1ktlrmv-2 cUADWe sticky-table-cell"
          >
            30
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
